### PR TITLE
Move the CUDA implementation of round to ATen.

### LINKED ISSUE #24617: Migrate `round` and `round_` from the TH to Aten (CUDA)
Porting TH operators is essential for code simplicity and performance reasons.

Porting guides and Q&A are available in umbrella issue: #24507

Feel free to add @VitalyFedyunin as a reviewer to get a prioritized review.

### PR DESCRIPTION
**Pull request by xuhdev**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25592 Simplify operator `sign` using the helper.
* #25423 Move the CUDA implementation of trunc to ATen.
* #25372 Move the CUDA implementation of floor to ATen.
* #25285 Move the CUDA implementation of rsqrt to ATen.
* **#25041 Move the CUDA implementation of round to ATen.**

Fix #24617

Pull Request resolved: https://github.com/pytorch/pytorch/pull/25041

Differential Revision: [D17114368](https://our.internmc.facebook.com/intern/diff/D17114368)

### CODE REVIEW

File: aten/src/ATen/native/cuda/UnaryOpsKernel.cu [hunk 1]
```diff
@@ -40,6 +40,14 @@ void ceil_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+void round_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "round_cuda", [&]() {
+    gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+      return std::round(a);
```
Comment (VitalyFedyunin) [inline aten/src/ATen/native/cuda/UnaryOpsKernel.cu]: See #17443 we had reasons to use `nearbyint`

**Review by VitalyFedyunin — CHANGES_REQUESTED** [review]
Overall good, but please use `nearbyint`

File: aten/src/ATen/native/cuda/UnaryOpsKernel.cu (hunk 1 above)
Comment (xuhdev) [reply aten/src/ATen/native/cuda/UnaryOpsKernel.cu]: I see, I have made the change and add a comment.

**Review by VitalyFedyunin — APPROVED** [review]

**Comment by VitalyFedyunin** [comment]
Can you please rebase, unrelated tests seems to be failing.

**Comment by xuhdev** [comment]
@VitalyFedyunin rebased

**Comment by xuhdev** [comment]
@VitalyFedyunin I've updated this stack in accordance to #25689

**Comment by xuhdev** [comment]
@VitalyFedyunin I have resolved the CI failures. The reason is that HIP cannot recognize std::nearbyint. I resolved this issue by using wrappers.

**Comment by xuhdev** [comment]
@VitalyFedyunin Is this ready for merge? It's been quite a while since this PR was made

**Pull request closed without merging**